azure-pipelines.yml

```diff
@@ -9,45 +9,36 @@ pr:
 - rel/*
 
 pool:
-  vmImage: windows-2022
+  vmImage: windows-latest
+
+variables:
+  BuildConfiguration: Release
+  BuildTools: $(Agent.ToolsDirectory)/dotnet
+  DotNetVersion: 7.0.x
 
 jobs:
 - job: BuildBits
+  displayName: Build and Test solution
   timeoutInMinutes: 60
   steps:
 
-  # Install the .NET Core 3.1 SDK
-  - task: UseDotNet@2
-    displayName: Install the .NET Core 3.1 SDK
-    inputs:
-      version: 3.1.x
-
-  # Install the .NET 6 SDK
-  - task: UseDotNet@2
-    displayName: Install the .NET 6 SDK
+  # Cache .NET SDKs and Tools across pipeline runs
+  - task: Cache@2
+    displayName: Cache .NET SDKs
     inputs:
-      version: 6.0.x
+      key: 'dotnet | "$(Agent.OS)" | "$(DotNetVersion)"'
+      restoreKeys: |
+        dotnet | "$(Agent.OS)"
+        dotnet
+      path: $(BuildTools)
 
-  # Install the .NET 7 SDK
+  # Install the .NET 7 SDK (Fixes dotnet/roslyn#63780)
```

Comment:
This is not needed to fix that but, it's needed to make the generators on the Roslyn 4.4 target work at all, as that version of the compiler is bundled with the .NET 7 SDK. I think we can drop this comment here to avoid confusion 🙂

Reply:
I'll update the comment. I thought it was because of the version of the `S.C.I.` library. I used the latest compiler with `PackageReference` mode and the error still occurred but installing .NET 7 SDK on my machine fixed the issue. So, it could be both.

Reply:
There's a couple different bugs in Roslyn 4.3, one is that issue with `S.C.I` you linked, and another is one that causes `ForAttributeWithMetadataName` to crash at runtime, which is why I dropped support for 4.3 and just jumped to 4.4. But that's already documented in the .csproj for the source generator, so it's not needed here. We'd still need the .NET 7 SDK anyway because I'll be adding a .NET 7 target to the HighPerformance shortly, so this bit in the pipeline doesn't need any comments, just like there were none when we were installing the .NET 6 one before 👍

Reply:
Then I'll drop the comment.